ktor-client/ktor-client-apache5/jvm/src/io/ktor/client/engine/apache5/Apache5Engine.kt

```diff
@@ -89,6 +89,8 @@ internal class Apache5Engine(override val config: Apache5EngineConfig) : HttpCli
                             ClientTlsStrategyBuilder.create()
                                 .setSslContext(config.sslContext ?: SSLContexts.createSystemDefault())
                                 .setTlsVersions(TLS.V_1_3, TLS.V_1_2)
+                                // TODO: Remove apply after https://github.com/apache/httpcomponents-client/pull/588 merged
+                                .apply { setHostnameVerificationPolicy(HostnameVerificationPolicy.BOTH) }
```

Comment:
This option seems the [safest one](https://github.com/apache/httpcomponents-client/blob/rel/v5.4/httpclient5/src/main/java/org/apache/hc/client5/http/ssl/HostnameVerificationPolicy.java). Tests fail with the default `BUILTIN` policy

Reply:
let's wait for the fix to be released

Reply:
good job on upstreaming the fix :)